File: Patterns/09. Diamond Patterns/84. Full Diamond Decreasing Numbers/fddn.java
import java.util.*;
public class fddn{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        int row=1;

        for(int i=1; i<=2*n-1; i++){
            if(i<=n){
                row = i;
            }else{
                row = 2*n-i;
            }
            // print spaces
            for(int j=1; j<=n-row; j++){
                System.out.print("  ");
            }
            // print first half
            for(int j=n; j>=n-row+1; j--){
                System.out.print(j+ " ");
            }
            // print second half
            for(int j=n-row+2; j<=n; j++){
                System.out.print(j + " ");
            }
            System.out.println();
        }
        
    }
}

/*
Output

        5 
      5 4 5 
    5 4 3 4 5 
  5 4 3 2 3 4 5 
5 4 3 2 1 2 3 4 5 
  5 4 3 2 3 4 5 
    5 4 3 4 5 
      5 4 5 
        5 

*/

File: Patterns/09. Diamond Patterns/84. Full Diamond Decreasing Numbers/fddn.py
n = int(input("Enter a Number : "))
row=1

for i in range(1,2*n):
    if(i<=n):
        row=i
    else:
        row=2*n-i
    
    # print spaces
    for j in range(1,n-row+1):
        print(end='  ')
    # print first half
    for j in range(n,n-row,-1):
        print(j, end=' ')
    # print second half
    for j in range(n-row+2,n+1):
        print(j, end=' ')
    print()

'''
Output:

        5 
      5 4 5 
    5 4 3 4 5 
  5 4 3 2 3 4 5 
5 4 3 2 1 2 3 4 5 
  5 4 3 2 3 4 5 
    5 4 3 4 5 
      5 4 5 
        5 
         
'''